pkg/apis/resource/helper.go

```diff
@@ -0,0 +1,243 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/model/environment"
+	"github.com/seal-io/walrus/pkg/dao/model/project"
+	"github.com/seal-io/walrus/pkg/dao/model/resourcedefinition"
+	"github.com/seal-io/walrus/pkg/dao/model/resourcedefinitionmatchingrule"
+	"github.com/seal-io/walrus/pkg/dao/model/templateversion"
+	"github.com/seal-io/walrus/pkg/dao/types/object"
+	"github.com/seal-io/walrus/pkg/dao/types/property"
+	"github.com/seal-io/walrus/pkg/deployer/terraform"
+	pkgenv "github.com/seal-io/walrus/pkg/environment"
+	"github.com/seal-io/walrus/pkg/resourcedefinitions"
+	"github.com/seal-io/walrus/utils/json"
+)
+
+// genComputedAttributes generate computed attributes for resource.
+func genComputedAttributes(
+	ctx context.Context,
+	entity *model.Resource,
+	modelClient *model.Client,
+) (property.Values, error) {
+	var computedAttrs property.Values
+
+	env, err := modelClient.Environments().Query().
+		Where(environment.ID(entity.EnvironmentID)).
+		WithProject(func(pq *model.ProjectQuery) {
+			pq.Select(
+				project.FieldID,
+				project.FieldName)
+		}).
+		Only(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
can we reuse the input validated result?

Reply:
Need query project name here.